CONTRIBUTING.rst

```diff
@@ -742,8 +495,56 @@ Resources & links
 
   - Airflow users mailing list: `<users-subscribe@airflow.apache.org>`_
 
-- `Issues on Apache’s Jira <https://issues.apache.org/jira/browse/AIRFLOW>`__
+- `Issues on Apache’s JIRA <https://issues.apache.org/jira/browse/AIRFLOW>`__
 
 - `Slack (chat) <https://apache-airflow-slack.herokuapp.com/>`__
 
+Step 4: Prepare PR
+------------------
+
+1. Update the local sources to address the JIRA ticket.
+
+   For example, to address this example JIRA ticket, do the following:
+
+   * Read about `email configuration in Airflow <https://airflow.readthedocs.io/en/latest/concepts.html#email-configuration>`__.
+
+   * Find the class you should modify. For the example ticket, this is `email.py <https://github.com/apache/airflow/blob/master/airflow/utils/email.py>`__.
+
+   * Find the test class where you should add tests. For the example ticket, this is `test_email.py <https://github.com/apache/airflow/blob/master/tests/utils/test_email.py>`__.
+
+   * Modify the class and add necessary code and unit tests.
+
+   * Run the unit tests from the `IDE <TESTING.rst#running-unit-tests-from-ide>`__ or `local virtualenv <TESTING.rst#running-unit-tests-from-local-virtualenv>`__ as you see fit.
+
+   * Run the tests in `Breeze <TESTING.rst#running-unit-tests-inside-breeze>`__.
+
+   * Run and fix all the `static checks <STATIC_CODE_CHECKS>`__. If you have
+     `pre-commits installed <STATIC_CODE_CHECKS.rst#pre-commit-hooks>`__,
+     this step is automatically run while you are committing your code. If not, you can do it manually
+     via ``git add`` and then ``pre-commit run``.
+
+2. Rebase your fork, squash commits, and resolve all conflicts.
+
+3. Re-run static code checks again.
+
+4. Create a pull request with the following title for the sample ticket:
+   [AIRFLOW-5934: Added extra CC: field to the Airflow emails].
```

Comment:
```suggestion
   ``[AIRFLOW-5934] Added extra CC: field to the Airflow emails.``
```

We should make a mention here about commit messages - that they should be short, but descriptive enough  to be useful in a changelog.

Reply:
We have this further down the list: 

> Adhere to guidelines for commit messages described in this `article <http://chris.beams.io/posts/git-commit/>`__.
>     This makes the lives of those who come after you a lot easier.

Reply:
Not in this document is it?

Also the suggestion is for correct balancing of `[]` :)

Reply:
Right :)

Reply:
@ashb:  It is in this document but in the lines that have not changed :). It's been there since I remember.